hack/charts/cluster-api-operator/templates/bootstrap-conditions.yaml

```diff
@@ -10,20 +10,21 @@ metadata:
     "helm.sh/hook-weight": "1"
   name: capi-system
 ---
-apiVersion: operator.cluster.x-k8s.io/v1alpha1
+apiVersion: operator.cluster.x-k8s.io/v1alpha2
 kind: CoreProvider
 metadata:
   name: cluster-api
   namespace: capi-system
   annotations:
     "helm.sh/hook": "post-install"
     "helm.sh/hook-weight": "2"
-{{- if .Values.secretName }}
+{{- with .Values.configSecret }}
 spec:
-  secretName: {{ .Values.secretName }}
-{{- if .Values.secretNamespace }}
-  secretNamespace: {{ .Values.secretNamespace }}
-{{- end }}
+  configSecret:
+    name: {{ .name }}
+    {{- if .namespace }}
+    namespace: {{ .namespace }}
+    {{- end }}
```

Comment:
while I have not tested this and all other changes regarding helm charts, just wanna raise a voice that we have had [issues](https://github.com/kubernetes-sigs/cluster-api-operator/pull/216) with it recently and if we tested templating changes locally at least?

Reply:
we only test provider deployment at the moment https://github.com/kubernetes-sigs/cluster-api-operator/blob/main/test/e2e/helm_test.go

Reply:
Yep, I tested this PR locally, plus we have some e2e tests in place.
The issue there happened because I put the object in the wrong section, so it's not strictly related... I wonder if we can introduce https://github.com/yannh/kubeconform to verify templates we generate.

Reply:
I’d recommend using chart tester for that purpose, it has support for linting and verification - https://github.com/helm/chart-testing

Reply:
ack, I'll add `chart-testing` support in another PR

Reply:
/lgtm